### PR TITLE
Research on hyperflux build

### PR DESCRIPTION
**Pull request by DanielBelmes**
Current TODO:

- Decide api design(can be pretty fluid)
- Decide external and peer deps

### CODE REVIEW

**Comment by DanielBelmes** [comment]
image
current build log snippet

File: packages/hyperflux/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+// });
+
+const productionConfig = {
+  minifyInternalExports: true,
+}
+function customBuild() {
+  return esbuild({minify: true})
+}
+
+const externals = ["react", "react-reconciler", "react-reconciler/constants","ts-matches", "ts-toolbelt", "uuid", "@hookstate/core", "@hookstate/identifiable"]
```
Comment (HexaField) [inline packages/hyperflux/rollup.config.js]: wonder if we can import this list from the package.json dependencies object keys

Comment (DanielBelmes) [reply packages/hyperflux/rollup.config.js]: This is just to suppress a warning. We don't really need it but I liked having the list.

Comment (DanielBelmes) [reply packages/hyperflux/rollup.config.js]: I am working on the vite config right now and for some reason it doesn't auto detect externals so I guess this is needed. Very weird

**Pull request closed without merging**